### PR TITLE
Working version 1.0.0

### PR DESCRIPTION
**Pull request by kalexrt**
This Pull request includes the following:
1. Added fixed chat backend and api
2. Improve prompts
3. Update readme
4. User chat history

### CODE REVIEW

File: nepal_constitution_ai/agent/agent.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+from nepal_constitution_ai.agent.utils import create_agent
+from langchain.tools import Tool
+
+def setup_agent(
+    retriever_chain,
+    conv_chain,
+    llm_model,
+):
```
Comment (amunamun) [inline nepal_constitution_ai/agent/agent.py:13]: type cast

File: nepal_constitution_ai/retriever/retriever_base.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            if len(self.chat_history.get_messages()) > 0:
+                new_query = rewrite_query(
+                    query=query, llm_model=self.llm_model, history=self.chat_history
+                )
+            else:
+                new_query = query
+            result = self.agent.invoke(
+                {"input": new_query}
+            )
+            output = result["output"]["answer"]
```
Comment (amunamun) [inline nepal_constitution_ai/retriever/retriever_base.py:59]: correct this

File: nepal_constitution_ai/retriever/utils.py [hunk 3]
```diff
@@ -16,19 +15,20 @@ def get_llm(llm_name: str) -> OpenaiModel:
...
 
+    Raises:
+        ValueError: If the provided LLM name is invalid.
     """
     try:
         llm_model = OpenaiModel(llm_name).model_selection()
-
     except Exception as e:
         logger.error(f"Invalid LLM model selected: {llm_name}")
         raise ValueError("Wrong llm model selected")
```
Comment (amunamun) [inline nepal_constitution_ai/retriever/utils.py:25]: correct this

File: nepal_constitution_ai/utils/utils.py [hunk 4]
```diff
@@ -1,3 +1,5 @@
+import uuid
+
 def parse_to_int(num: str) -> int:
```
Comment (amunamun) [inline nepal_constitution_ai/utils/utils.py]: use Union[int, None]

**Comment by Znis** [comment]
Made changes as requested on the comments
Rest looks good to me

**Review by Znis — APPROVED** [review]
Looks good to me

**Pull request merged**